fake_spectra/spectra.py

```diff
@@ -606,11 +613,30 @@ def replace_not_DLA(self, ndla, thresh=10**20.3, elem="H", ion=1):
         #Declare variables
         found = 0
         wanted = ndla
+        size_ind = None
         cofm_DLA = np.empty_like(self.cofm)[:ndla, :]
         #Filter
         #Note: line does nothing
         col_den = self.compute_spectra(elem,ion,1215,False)
-        ind = self.filter_DLA(col_den, thresh)
+        cdsum=np.sum(col_den, axis=1)
+        # A variable for comm.Reduce()
+        cdsum_added = np.zeros_like(cdsum, dtype='d')
```

Comment:
Delete the dtype. If you use zeros_like it will automatically have the same type as the parent.

Reply:
Yes, you are right. I have deleted at some point, the issue above has not been solved. However, you are right , that is redundant and I will delete that.

Reply:
Yes, I agree that it won't solve this issue. But if you were to change the type of cdsum in future it could cause a new bug.